File: string/Longest Substring Without Repeating Characters.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int length_string = s.length();
        set<char>set_string;
        int ret_length = 0;
        int i = 0, j =0;
        while(i<length_string && j<length_string)
        {
           if(set_string.count(s[j]) == 0)
               {
                   set_string.emplace(s[j]);
                    j++;
                   ret_length = max(ret_length,j-i);
                   
               }
              else
               {
                   set_string.erase(s[i]);
                   i++;
               }
        }
        return ret_length;
        
        
    }
};

File: String/longest_substring_without_repeating_characters.py
class Solution(object):
    def lengthOfLongestSubstring(self, s):
        """
        :type s: str
        :rtype: int
        """
        if len(s) == 0:
            return 0
        temp = ""
        result = 0
        for char in s:
            if char not in temp:
                temp += char
                
            else:
                print("char:",char)
                char_index = temp.index(char) + 1
                print("char_index:",char_index)
                temp = temp[char_index:]
                print("temp:",temp)
                temp += char
            
            result = max(result,len(temp))
        
        return result
        
